src/Passable.js

```diff
@@ -1,134 +1,67 @@
+class Passable {
 
-        if (typeof args[0] === 'string') {
-            // warn rather than fail
-            severity = args[0] === WARN ? WARN : FAIL;
-        }
+    constructor(name, ...args) {
+        const computedArgs = passableArgs(args);
+        this.specific = computedArgs.specific;
+        this.custom = computedArgs.custom;
 
-        if (!testsPerformed.hasOwnProperty(dataName)) {
-            testsPerformed[dataName] = {
-                testCount: 0,
-                failCount: 0,
-                warnCount: 0
-            };
-        }
+        this.res = initResponseObject(name);
 
-        if (!isValid) {
-            // on failure/error, bump up the counters
-            severity === FAIL ? onError(dataName, statement) : onWarn(dataName, statement);
-        }
+        this.pass = this.pass.bind(this);
+        this.enforce = this.enforce.bind(this);
 
-        // bump overall counters
-        testsPerformed[dataName].testCount++;
-        testCount++;
+        if (typeof computedArgs.passes === 'function') {
+            computedArgs.passes(this.pass, this.enforce);
+        }
 
-        return isValid;
-    };
-
-    function onError(dataName, statement) {
-        hasValidationErrors = true;
-        validationErrors[dataName] = validationErrors[dataName] || [];
-        validationErrors[dataName].push(statement);
-        failCount++;
-        testsPerformed[dataName].failCount++;
+        return this.res;
     }
 
-    function onWarn(dataName, statement) {
-        hasValidationWarnings = true;
-        validationWarnings[dataName] = validationWarnings[dataName] || [];
-        validationWarnings[dataName].push(statement);
-        warnCount++;
-        testsPerformed[dataName].warnCount++;
+    enforce(value) {
+        const e = enforce.bind(this.currentPass);
+        return e(value, this.custom);
     }
 
-    function generateResultObject() {
-        const result = {
-            name,
-            hasValidationErrors,
-            hasValidationWarnings,
-            testsPerformed,
-            validationErrors,
-            validationWarnings,
-            failCount,
-            warnCount,
-            testCount,
-            skipped
-        };
-
-        return result;
-    }
+    pass(fieldName, statement, ...args) {
 
-    if (typeof passes === 'function') {
-        // register all the tests
-        const Enforce = (value) => {
-            const e = enforce.bind(currentPass);
-            return e(value, custom);
-        };
+        this.currentPass = {};
 
-        // run all units in the group
-        passes(pass, Enforce);
+        if (this.specific.length && this.specific.indexOf(fieldName) === -1) {
+            this.res.skipped.push(fieldName);
+            return;
+        }
+
+        this.res.testsPerformed[fieldName] = this.res.testsPerformed[fieldName] || initField();
+
+        // callback is always the last argument
+        const callback = args.slice(-1)[0],
```

Comment:
It's kind of tricky with the optional argument in one-before-last position. The interface might be clearer if you pop the callback:
```javascript
const callback = args.pop();
const severity = args[0] || FAIL;
```
What do you think?